pkg/apis/apis/v1alpha1/crd_to_apiresourceschema.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// CRDToAPIResourceSchema converts a CustomResourceDefinition to an APIResourceSchema. The name of the returned
+// APIResourceSchema is in the form of <prefix>.<crd.Name>.
+func CRDToAPIResourceSchema(crd *apiextensionsv1.CustomResourceDefinition, prefix string) (*APIResourceSchema, error) {
+	name := prefix + "." + crd.Name
+
+	if msgs := validation.IsDNS1123Subdomain(name); len(msgs) > 0 {
+		var errs []error
+
+		for _, msg := range msgs {
+			errs = append(errs, field.Invalid(field.NewPath("metadata", "name"), name, msg))
+		}
+
+		return nil, errors.NewAggregate(errs)
+	}
+
+	apiResourceSchema := &APIResourceSchema{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: SchemeGroupVersion.String(),
+			Kind:       "APIResourceSchema",
```

Comment:
little uncommon to set these. Should be done where we serializer this.

Reply:
Yeah, I was using yaml.Marshal before. Will remove. Thanks!

Reply:
Fixed in force-push